0012-BIRS-distributed-scheduler.md

```diff
@@ -0,0 +1,463 @@
+
+```Json
+{
+  "name": "prd-db-backup",
+  "schedule": "@daily",
+  "data": {
+    "task": "db-backup",
+    "metadata": {
+      "db_name": "my-prod-db",
+      "backup_location": "/backup-dir"
+    }
+  }
+}
+```
+
+Potential `dapr/go-sdk` example code:
+```go
+import(
+    ...
+    scheduler "github.com/dapr/dapr/pkg/proto/scheduler/v1"
+    ...
+)
+...
+
+type Metadata struct {
+    DBName         string `json:"db_name"`
+    BackupLocation string `json:"backup_location"`
+}
+
+type DbBackup struct {
+    Task     string   `json:"task"`
+    Metadata Metadata `json:"metadata"`
+}
+
+func main() {
+...
+
+    // Define a job to be scheduled
+    job := &scheduler.Job{
+        Name:     "prd-db-backup",
+        Schedule: "@daily",
+        Data: &ptypes.Any{
+            Value: &DbBackup{
+                Task: "db-backup",
+                Metadata: Metadata{
+                    DBName:         "my-prod-db",
+                    BackupLocation: "/backup-dir",
+                },
+            },
+        },
+    }
+
+    // Schedule a job
+	scheduleJobRequest := &scheduler.ScheduleJobRequest{
+		Job:          job,
+	}
+
+    _, err = client.ScheduleJob(context.Background(), scheduleJobRequest)
+	if err != nil {
+		fmt.Printf("Error scheduling job: %v\n", err)
+	}
+
+	// Get a job by name
+	getJobRequest := &scheduler.GetJobRequest{
+		Name: "prd-db-backup",
+	}
+
+	response, err := client.GetJob(context.Background(), getJobRequest)
+	if err != nil {
+		fmt.Printf("Error getting job: %v\n", err)
+	} else {
+		job := response.Job
+		fmt.Printf("Got job: %v\n", job)
+	}
+
+	// List all jobs by app_id
+	listJobsRequest := &scheduler.ListJobsRequest{
+		AppID: "your-app-id",
+	}
+
+	listResponse, err := client.ListJobs(context.Background(), listJobsRequest)
+	if err != nil {
+		fmt.Printf("Error listing jobs: %v\n", err)
+	} else {
+		jobs := listResponse.Jobs
+		fmt.Printf("List of jobs: %v\n", jobs)
+	}
+
+	// Delete a job by name
+	deleteJobRequest := &scheduler.DeleteJobRequest{
+		Name: "prd-db-backup",
+	}
+
+	_, err = client.DeleteJob(context.Background(), deleteJobRequest)
+	if err != nil {
+		fmt.Printf("Error deleting job: %v\n", err)
+	}
+...
+}
+```
+
+##### Actor Reminders
+
+The interval functionality of the Actor Reminder is similar to the job schedule. With Actor Reminders, a user can specify:
+```json
+{
+    "dueTime": "10s",
+    "period": "R4/PT3S",
+    "ttl": "10s"
+}
+```
+
+Similar logic can be applied to a job in the following manner: 
+
+- `dueTime` => A job is expected to be scheduled as soon as it is submitted. There is no delaying a job to be scheduled, so this feature will not be supported for jobs
```

Comment:
At the time, I didn't see a need for `dueTime` given a job `schedule`. If there is a need for `dueTime` to support workflows, that the job `schedule` doesn't account for, then this can be added to the Job spec 👍🏻 

I could see wanting to use `dueTime` of waiting a month before the job will fire, and then using the `schedule` as an interval. Is that a similar scenario to what you were thinking for Workflows?

Reply:
Yes, the addition of `dueTime` to do things like wait a month is exactly what we'd need for workflows. In the happy path for workflows, we'd never actually need to do repeated reminders. It's the ability to schedule a single reminder in the future which is critical.

Reply:
+1 for adding `dueTime` -  It's a slightly different use-case to that of a `schedule` but just as important IMO, particularly for Line of Business use-cases.

Also, delayed message delivery in Azure Service Bus use an explicit UTC date time for when a message should be attempted for delivery, rather than a schedule. [example](https://learn.microsoft.com/en-us/dotnet/api/azure.messaging.servicebus.servicebusmessage.scheduledenqueuetime?view=azure-dotnet)